pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -126,7 +126,13 @@ if [ -n "$flake" ]; then
 fi
 
 if [ -n "$flake" ]; then
-    flake=$(nix "${flakeFlags[@]}" flake info --json "${extraBuildFlags[@]}" "${extraLockFlags[@]}" -- "$flake" | jq -r .url)
+    if nix flake metadata &>/dev/null; then
```

Comment:
This doesn’t work unless you have `experimental-features = nix-command flakes` in your `nix.conf`.
```suggestion
    if nix "${flakeFlags[@]}" flake metadata &>/dev/null; then
```

Reply:
Also… this actually only works if you're in a directory that is actually a flake itself. How about `nix "${flakeFlags[@]}" flake metadata --version &>/dev/null`?

Reply:
Excellent! Thanks!

Reply:
@emilazy I just tested it with your proposed changes in my own dotfiles build pipeline. I'm moving `nix.conf` before switching. It works, whereas it didn't before I added your change. Thanks for pointing that out!